File: LL/src/Presentation/ll/src/app/core/services/real-time/audience/audience.ts
export type Audience = CharacterAudience | GuildAudience | WorldAudience;

export interface CharacterAudience {
  type: 'character';
  characterId: string; // Guid as string
}

export interface GuildAudience {
  type: 'guild';
  guildId: string; // Guid as string
}

export interface WorldAudience {
  type: 'world';
}

File: LL/src/Core/Application/Interfaces/WebSockets/Audience.cs
namespace Application.Interfaces.WebSockets;
public abstract record Audience
{
    public sealed record Character(Guid CharacterId) : Audience;
    public sealed record Characters(IReadOnlyCollection<Guid> CharacterIds) : Audience;
    public sealed record Guild(Guid GuildId) : Audience;
    public sealed record Raid(Guid RaidRunId) : Audience;
    public sealed record TournamentGrounds : Audience;
    public sealed record World : Audience;
}
